test-suite/shared/src/test/scala/org/scalajs/testsuite/scalalib/ArrayBuilderTest.scala

```diff
@@ -0,0 +1,261 @@
+
+    assertSame(classOf[Array[Char]], a.getClass)
+    assertEquals(2, a.length)
+    assertTrue(erase(a(0)).isInstanceOf[Char])
+    assertTrue(erase(a(0)).equals('A'))
+    assertTrue(erase(a(1)).equals('S'))
+  }
+
+  @Test def Char_zeros_inline(): Unit = {
+    val a = zerosInline[Char](3)
+    assertSame(classOf[Array[Char]], a.getClass)
+    assertEquals(3, a.length)
+    assertTrue(erase(a(0)).isInstanceOf[Char])
+    assertTrue(erase(a(0)).equals('\0'))
+  }
+
+  @Test def Char_zeros_noinline(): Unit = {
+    val a = zerosNoInline[Char](3)
+    assertSame(classOf[Array[Char]], a.getClass)
+    assertEquals(3, a.length)
+    assertTrue(erase(a(0)).isInstanceOf[Char])
+    assertTrue(erase(a(0)).equals('\0'))
+  }
+
+  @Test def Boolean_normal_case_inline(): Unit = {
+    val b = ArrayBuilder.make[Boolean]()
+    b += true
+    b += someBoolean
+    val a = b.result()
+
+    assertSame(classOf[Array[Boolean]], a.getClass)
+    assertEquals(2, a.length)
+    assertTrue(erase(a(0)).isInstanceOf[Boolean])
+    assertTrue(erase(a(0)).equals(true))
+    assertTrue(erase(a(1)).equals(false))
+  }
+
+  @Test def Boolean_normal_case_noinline(): Unit = {
+    val b = makeNoInline[Boolean]()
+    b += true
+    b += someBoolean
+    val a = b.result()
+
+    assertSame(classOf[Array[Boolean]], a.getClass)
+    assertEquals(2, a.length)
+    assertTrue(erase(a(0)).isInstanceOf[Boolean])
+    assertTrue(erase(a(0)).equals(true))
+    assertTrue(erase(a(1)).equals(false))
+  }
+
+  @Test def Boolean_zeros_inline(): Unit = {
+    val a = zerosInline[Boolean](3)
+    assertSame(classOf[Array[Boolean]], a.getClass)
+    assertEquals(3, a.length)
+    assertTrue(erase(a(0)).isInstanceOf[Boolean])
+    assertEquals(false, erase(a(0))) // scalastyle:ignore
+  }
+
+  @Test def Boolean_zeros_noinline(): Unit = {
+    val a = zerosNoInline[Boolean](3)
+    assertSame(classOf[Array[Boolean]], a.getClass)
+    assertEquals(3, a.length)
+    assertTrue(erase(a(0)).isInstanceOf[Boolean])
+    assertEquals(false, erase(a(0))) // scalastyle:ignore
+  }
+
+  @Test def Unit_normal_case_inline(): Unit = {
+    val b = ArrayBuilder.make[Unit]()
+    b += (())
+    val a = b.result()
+
+    assertSame(classOf[Array[Unit]], a.getClass)
+    assertEquals(1, a.length)
+    assertTrue(erase(a(0)).isInstanceOf[Unit])
+    assertTrue(erase(a(0)).equals(()))
+  }
+
+  @Test def Unit_normal_case_noinline(): Unit = {
+    val b = makeNoInline[Unit]()
+    b += (())
+    val a = b.result()
+
+    assertSame(classOf[Array[Unit]], a.getClass)
+    assertEquals(1, a.length)
+    assertTrue(erase(a(0)).isInstanceOf[Unit])
+    assertTrue(erase(a(0)).equals(()))
+  }
+
+  @Test def Unit_zeros_inline(): Unit = {
+    val a = zerosInline[Unit](3)
+    assertSame(classOf[Array[Unit]], a.getClass)
+    assertEquals(3, a.length)
+    if (!executingInJVM) {
+      assertTrue(erase(a(0)).isInstanceOf[Unit]) // JVM says false
+      assertEquals((()), erase(a(0))) // JVM says false
+    }
```

Comment:
I suggest adding an `else` branch, asserting that the JVM indeed says false. (and then remove the comments as it will be obvious)